include/swift/AST/ASTContext.h

```diff
@@ -255,6 +261,9 @@ class ASTContext {
   /// Optional table of counters to report, nullptr when not collecting.
   UnifiedStatsReporter *Stats = nullptr;
 
+  /// If non-null, the YAML file where remarks should be recorded.
+  std::unique_ptr<llvm::yaml::Output> OptimizationRecordFile;
```

Comment:
Please don't put this on the ASTContext. If it has to live somewhere that everyone has access to it, SILPassManager or SILModule is a better place; please ask someone on Erik's team what they think.

Reply:
@eeckstein, @gottesmm do you guys have a suggestion where this should go instead of ASTContext.  This is the file stream where optimizations insert their remarks in YAML format.  Should it be in SILModule or SILPassManager or somewhere else?

Reply:
I haven't thought about it completely, but it does feel like a SILModule thing. Or perhaps a SILOption? Not sure.

Reply:
SILOptions seems to be a good candidate. You can access the SILOptions from the SILModule and SILPassManager

Reply:
We recently ran into trouble putting a std::unique_ptr in FrontendOptions because it's copied. I suspect SILOptions will have the same issue.

Reply:
Thanks @eeckstein, I was gravitating toward SILOptions too based on @gottesmm's suggestion as putting it into SILModule turned out to be a bit too convoluted:

(1) SILModule is constructed a few levels down from performCompile and it gets messy to properly pass down more things to its ctor so I had to use a setter which would have been less nice.
(2) The code is also less clear since we need to call keep() on the generated YAML file if no error has occurred in performCompile() even though the file is created somewhere else.

This should be more clear with SILOptions.

@jrose-apple, yep :(, this thing again...